### PR TITLE
feat: HACS installation support

### LINKED ISSUE #30: HACS repository support
**Is your feature request related to a problem? Please describe.**
HACS simplifies a lot the deployment and updating of out-of-tree integrations in hassio.

**Describe the solution you'd like**
I would like for this integration to add HACS support.

**Additional context**
I'll provide a PR to add this feature

### PR DESCRIPTION
**Pull request by gervaso**
### Related Issues

- Closes #30 

### Proposed Changes:
I created a basic hacs.json file in the root folder and a small modification to the build and release workflow to create and publish a zip file tied to any tagged release.

### Testing:
After publishing several releases on my repo I manually added it to the HACS "custom repositories" and installed it on my system where it worked as expected.

### Extra Notes (optional):
This will make installing and updating the component much easier for any HACS user.
Once it will be used and tested by enough people it could be worthy asking to be added to the "main" HACS repo list.

### Checklist

- [x] Related issues and proposed changes are filled
- [x] Tests are defining the correct and expected behavior
- [x] Code is well-documented via docstrings

### CODE REVIEW

File: .github/workflows/building.yaml [hunk 1]
```diff
@@ -7,7 +7,7 @@ on:
       - 'v*'
 
 permissions:
-  contents: read
```
Comment (palazzem) [inline .github/workflows/building.yaml:10]: I think this action to publish the package in HACS (so the ZIP build) may live in its own GitHub action. What do you think if we have something along these lines:
- `workflows/building.yaml` that includes the building step for a regular package (as it is right now), so that who doesn't use HACS, simply uses this package.
- `workflows/hacs.yaml` that is basically your change!

While both are actually "building a package", this one has the extra step of publishing it. I think it may be better to separate it so that you can change the trigger at the `publish` step instead of `push tags`.

Long story short, what if your change creates `hacs.yaml` that can be inspired by what the [HACS official package](https://github.com/hacs/integration/blob/main/.github/workflows/publish.yml) does?

In this way, instead of triggering this action at `v*` tag, you trigger it when we publish a new package: https://github.com/hacs/integration/blob/main/.github/workflows/publish.yml#L3-L6

File: .github/workflows/building.yaml [hunk 2]
```diff
@@ -52,3 +52,13 @@ jobs:
           name: econnect-alarm-${{ env.PKG_VERSION }}.tar.gz
           path: dist/econnect_alarm-${{ env.PKG_VERSION }}.tar.gz
           if-no-files-found: error
+
+      - name: ZIP Component Dir
```
Comment (palazzem) [inline .github/workflows/building.yaml:56]: nitpick: we may call this step "Build Integration ZIP package"

File: .github/workflows/building.yaml [hunk 3]
```diff
@@ -52,3 +52,13 @@ jobs:
...
+        run: |
+          cd ${{ github.workspace }}/custom_components/econnect_alarm
+          zip econnect_alarm.zip -r ./
+
+      - name: Upload zip to release
+        uses: softprops/action-gh-release@v0.1.15
+        with:
+          files: ${{ github.workspace }}/custom_components/econnect_alarm/econnect_alarm.zip
```
Comment (palazzem) [inline .github/workflows/building.yaml:64]: instead of calling the file `econnect_alarm.zip` we may call it `hacs_econnect_alarm.zip`. Just to mention people that this is a HACS installer package. What do you think?

File: hacs.json [hunk 4]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "Home Assistant e-Connect Integration (Elmo)",
+    "content_in_root": false,
+    "render_readme": true,
+    "hide_default_branch": true,
+    "zip_release": true,
+    "filename": "econnect_alarm.zip"
```
Comment (palazzem) [inline hacs.json:7]: If you agree with the previous comment, we should change also this.

File: .github/workflows/building.yaml [hunk 5]
```diff
@@ -52,3 +52,13 @@ jobs:
           name: econnect-alarm-${{ env.PKG_VERSION }}.tar.gz
           path: dist/econnect_alarm-${{ env.PKG_VERSION }}.tar.gz
           if-no-files-found: error
+
+      - name: ZIP Component Dir
+        run: |
+          cd ${{ github.workspace }}/custom_components/econnect_alarm
+          zip econnect_alarm.zip -r ./
+
+      - name: Upload zip to release
```
Comment (palazzem) [inline .github/workflows/building.yaml:61]: This step is good if we manage to trigger it when the event is `publish`. I think you can find more details in the [HACS repository](https://github.com/hacs/integration/blob/main/.github/workflows/publish.yml#L61)!

Other than that, I would recommend to still publish the artifact, so that we can attach it to the run that generated it. It's useful for debug purposes. This is an example from this file: https://github.com/palazzem/ha-econnect-alarm/blob/68e39bfd05f88ce8c6e2d10f134c68f80d73d41c/.github/workflows/building.yaml#L49-L54

**Review by palazzem — CHANGES_REQUESTED** [review]
Thank you very much for your first contribution! This is a great idea we should merge to improve the quality and discovery of the integration! I left few comments that I think may make your change even better!

To summarize, I think if you create a separated `hacs.yaml` is better, as you can trigger the action not when we tag, but when we release. After you make the change, I can do another pass to get it merged today!

File: .github/workflows/building.yaml (hunk 1 above)
Comment (gervaso) [reply .github/workflows/building.yaml]: I agree on all the requests but one, so I went forward and added a separate file

File: .github/workflows/building.yaml (hunk 5 above)
Comment (gervaso) [reply .github/workflows/building.yaml]: So far this is just a zipped version of the content of "custom_components", so I'll leave it this way and eventually leverage the "download source" if needed.

Comment (palazzem) [reply .github/workflows/building.yaml]: Sure thing!

**Review by palazzem — APPROVED** [review]
This is great! I'm going to merge this PR upstream as soon as we release the current pending version. So we can ship this change with a dedicated release!

Thank you very much for your contribution!
